### ISSUE TITLE
My Site Dashboard (Phase 2): Dashboard Cards

### ISSUE THREAD

**Issue by momo-ozawa** [wordpress-mobile/WordPress-iOS#17873]
## Description

General BlogDashboardViewController tasks for phase 2 of the My Site Dashboard project.

Ref: pbArwn-3mP-p2

## Tasks

- [x] Add new cards endpoint to WPKit https://github.com/wordpress-mobile/WordPressKit-iOS/pull/485
- [x] Call the new cards endpoint #17889
- [x] Parse the cards response #17909
- [x] Improve how the response is injected into the cards #17951
- [x] Handle offline support #17945
- [x] Handle switching blogs via the SitePickerVC #17935
- [x] Handle auto-updating (see: Android implementation) #18016
- [x] Handle pull-to-refresh #18025
- [x] Add pull-to-refresh events #18033
- [x] Hook up FAB actions for BlogDashboardViewController #18026
- [x] Load the local cards first and show loading indicator #18060
- [x] Handle failures and stale data #18068
- [x] Add wpComID to DashboardCardModel #18087 
- [x] Remove provisional loading indicator #18088
- [x] Improve performance (see: [comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/17921#pullrequestreview-877877396)) #18092
- [x] Check site creation flow (see: [comment](https://github.com/wordpress-mobile/WordPress-iOS/pull/17943#issuecomment-1036439504)) #18096
- [x] Fix BlogDetailsViewController instantiation #17949
- [x] Fix breaking constraints #18111
- [x] Fix iPad default selected cell #18118 
- [x] Add AB logic to display dashboard for half of new users (existing users should always see the Site Menu as the default selection) #18125
- [x] [Add track events and change where the AB experiment is assigned](https://github.com/wordpress-mobile/WordPress-iOS/issues/18150) #18157

### Descoped

- [ ] Give the user a feedback that the dashboard is being updated ([see comment](https://github.com/wordpress-mobile/WordPress-iOS/issues/17873#issuecomment-1058306079))

### Additional information: Improve how the response is injected into the cards

Right now, we're using `NSDictionary` to inject the data into the card's cell. The problem with this approach is that dealing with dictionaries creates a lot of boilerplate code and the cell has to parse it at the moment of configuration. Ideally, we should use a `Decodable` entity and inject it into the cells. However, there are a few challenges to using `Decodable` + inject the data into each cell.

**Comment by leandroalonso** [comment]
@momo-ozawa FYI: I re-ordered the tasks in order of priority.

**Comment by leandroalonso** [comment]
Regarding the task: "Give the user feedback that the dashboard is being updated" I worked on a proof of concept and shared it with @osullivanchris:

https://user-images.githubusercontent.com/7040243/156619455-4f0f6357-522d-4f8c-bef9-aab2745421e1.mp4

The results looked more like a glitch than feedback.

For now, we decided to only show this if the user actually pulls to refresh. We'll wait to see if any feedback is necessary based on usage and, if needed, we'll iterate.
